### PR TITLE
Emission conditions-of-sale support

### PR DESCRIPTION
**Pull request by ashfurrow**
This PR, made into [this existing PR](https://github.com/artsy/eigen/pull/2586), adds support for Emission components to properly display `/conditions-of-sale` routes. I made this PR targeting that branch because it makes it clear what I've changed; in reality, we'll probably wait for #2586 to be merged and then change this PR to target `master`.

Work tracked against [PURCHASE-60](https://artsyproduct.atlassian.net/browse/PURCHASE-60). There will be a corresponding Emission PR shortly.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/App/ARSwitchBoard.m [hunk 1]
```diff
@@ -200,6 +200,16 @@ - (void)updateRoutes
         return [[ARNavigationController alloc] initWithRootViewController:submissionVC];
     }];
 
+    [self.routes addRoute:@"/conditions-of-sale" handler:JLRouteParams {
+        // We want to fall back to the default routing unless this query parameter is specified, from Emission.
```
Comment (alloy) [inline Artsy/App/ARSwitchBoard.m]: Can this comment elaborate on why we want that?

File: Artsy/App/ARAppDelegate+Emission.m [hunk 2]
```diff
@@ -181,9 +181,18 @@ - (void)setupSharedEmissionWithPackagerURL:(NSURL *)packagerURL;
...
+            // and BidFlow are presented modally, and we want to let them present modal view controllers on top of themselves.
+            if ([[ARTopMenuViewController sharedController] presentedViewController]) {
+                [[[ARTopMenuViewController sharedController] presentedViewController] presentViewController:viewController
+                                                                                                   animated:ARPerformWorkAsynchronously
+                                                                                                 completion:nil];
+            } else {
+                [[ARTopMenuViewController sharedController] presentViewController:viewController
+                                                                         animated:ARPerformWorkAsynchronously
+                                                                       completion:nil];
+            }
```
Comment (alloy) [inline Artsy/App/ARAppDelegate+Emission.m]: Tiny nit: you could capture the correct presenting VC in a var first and then have only 1 copy of the `presentViewController:animated:completion:` message.

File: Artsy/App/ARSwitchBoard.m (hunk 1 above)
Comment (ashfurrow) [reply Artsy/App/ARSwitchBoard.m]: Certainly, will do.

File: Artsy/App/ARAppDelegate+Emission.m (hunk 2 above)
Comment (ashfurrow) [reply Artsy/App/ARAppDelegate+Emission.m]: Sure thing.

**Comment by alloy** [comment]
Ace, thanks 👍

**Comment by ashfurrow** [comment]
I've targeted this at the `emission-1.5.x-develop` branch instead, so we are good to merge whenever 👍

File: CHANGELOG.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ upcoming:
     - WIP Buy now works show in an Auction screen - Chris
     - ARVIR fixes & improvements - orta
     - Buy now layout adjustments, 3 columns on ipad - maxim
+    - Conditions-of-sale routing works - ash
```
Comment (alloy) [inline CHANGELOG.yml]: This should go to a new section, but feel free to merge as-is and do that in a follow up PR.

Comment (ashfurrow) [reply CHANGELOG.yml]: I agree it should go in a new section, but this is in the "upcoming" section and there's only one 😄 All this juggling eh? Thoughts?

**Comment by orta** [comment]
This should get merged on green, that branch can deal with the CHANGELOG changes between versions

**Comment by alloy** [comment]
Agreed, did not realise this wasn’t merged yet, sorry about that.

**Pull request merged**